geode-wan/src/upgradeTest/java/org/apache/geode/cache/wan/WANRollingUpgradeNewSenderProcessOldEvent.java

```diff
@@ -0,0 +1,411 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.wan;
+
+import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.distributed.internal.InternalLocator;
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.internal.cache.wan.parallel.ParallelGatewaySenderQueue;
+import org.apache.geode.test.dunit.DistributedTestUtils;
+import org.apache.geode.test.dunit.Host;
+import org.apache.geode.test.dunit.NetworkUtils;
+import org.apache.geode.test.dunit.VM;
+
+public class WANRollingUpgradeNewSenderProcessOldEvent
+    extends WANRollingUpgradeDUnitTest {
+
+  /**
+   * 1.create a sender, pause
+   * 2.do puts, build up the queue.
+   * 3.roll one server to current and stop another old server
+   * 4.start sender on current
+   * Current sender should handle old events, the remove site should receive all of them
+   */
+  @Test
+  public void oldEventShouldBeProcessedAtNewSender() {
```

Comment:
I like your comment for this function. I shows the 4 distinct steps of this method.

Maybe each step can be pulled out into it's own function since this function is already over 100 lines. It might help with readability

Reply:
Same with `oldEventShouldBeProcessedAtTwoNewSender`. Maybe a few helper functions can be created that can be used in both `oldEventShouldBeProcessedAtTwoNewSender` and `oldEventShouldBeProcessedAtNewSender`